rootston/output.c

```diff
@@ -95,36 +103,55 @@ static void render_surface(struct wlr_surface *surface,
 static void render_xdg_v6_popups(struct wlr_xdg_surface_v6 *surface,
 		struct roots_desktop *desktop, struct wlr_output *wlr_output,
 		struct timespec *when, double base_x, double base_y, float rotation) {
-	// TODO: make sure this works with view rotation
+	double width = surface->surface->current->width;
+	double height = surface->surface->current->height;
+
 	struct wlr_xdg_surface_v6 *popup;
 	wl_list_for_each(popup, &surface->popups, popup_link) {
 		if (!popup->configured) {
 			continue;
 		}
 
-		double popup_x = base_x + surface->geometry->x +
-			popup->popup_state->geometry.x - popup->geometry->x;
-		double popup_y = base_y + surface->geometry->y +
-			popup->popup_state->geometry.y - popup->geometry->y;
-		render_surface(popup->surface, desktop, wlr_output, when, popup_x,
-			popup_y, rotation);
-		render_xdg_v6_popups(popup, desktop, wlr_output, when, popup_x, popup_y,
-			rotation);
+		double popup_width = popup->surface->current->width;
+		double popup_height = popup->surface->current->height;
+
+		double popup_x = surface->geometry->x + popup->popup_state->geometry.x -
+			popup->geometry->x;
+		double popup_y = surface->geometry->y + popup->popup_state->geometry.y -
+			popup->geometry->y;
```

Comment:
I think this calculation is complicated enough that it needs to be in the wlr-xdg implementation.

Reply:
Yeah, will do.